dali/operators/image/resize/tensor_resize_attr.cc

```diff
@@ -0,0 +1,288 @@
+  } else {
+    assert(false);  // should not happen
+  }
+
+  if (has_roi_) {
+    GetShapeLikeWithAxes<float>(roi_start_, roi_start_arg_, spec, ws, "roi_start", axes, nsamples,
+                                spatial_ndim_, nargs, first_spatial_dim_);
+    GetShapeLikeWithAxes<float>(roi_end_, roi_end_arg_, spec, ws, "roi_end", axes, nsamples,
+                                spatial_ndim_, nargs, first_spatial_dim_);
+  }
+
+  max_size_.resize(spatial_ndim_,
+                   std::nextafter(static_cast<float>(std::numeric_limits<int>::max()), 0.0f));
+  if (has_max_size_) {
+    GetSingleOrRepeatedArg(spec, max_size_arg_, "max_size", nargs);
+    for (int i = 0; i < nargs; i++) {
+      int d = axes_[i] - first_spatial_dim_;
+      assert(d >= 0);
+      assert(d < spatial_ndim_);
+      max_size_[d] = max_size_arg_[i];
+    }
+  }
+
+  SmallVector<float, 3> requested_size, in_lo, in_hi;
+  requested_size.resize(spatial_ndim_);
+  in_lo.resize(spatial_ndim_);
+  in_hi.resize(spatial_ndim_);
+
+  assert(has_sizes_ + has_scales_ == 1);
+  const float *max_size = has_max_size_ ? max_size_.data() : nullptr;
+  for (int i = 0; i < nsamples; i++) {
+    auto in_sample_shape = input_shape.tensor_shape_span(i);
+    if (has_sizes_) {
+      for (int d = 0; d < spatial_ndim_; d++) {
+        requested_size[d] = sizes_[i * ndim_ + first_spatial_dim_ + d];
+      }
+    } else {
+      assert(has_scales_);
+      for (int d = 0; d < ndim_; d++) {
+        requested_size[d] = scales_[i * ndim_ + first_spatial_dim_ + d] * in_sample_shape[d];
+      }
+    }
+
+    bool empty_input = volume(input_shape.tensor_shape_span(i)) == 0;
+    CalculateInputRoI(in_lo, in_hi, has_roi_, roi_relative_, make_cspan(roi_start_),
+                      make_cspan(roi_end_), input_shape, i, spatial_ndim_, first_spatial_dim_);
+
+    span<const float> alignment;
+    if (has_alignment_)
+      alignment = {alignment_.data() + i * spatial_ndim_, spatial_ndim_};
+    assert(subpixel_scale_ || !has_alignment_);
+    CalculateSampleParams(params_[i], requested_size, in_lo, in_hi, subpixel_scale_, empty_input,
+                          spatial_ndim_, mode_, max_size, alignment, scale_round_fn_);
+  }
+
+  auto unchanged_dim = [&](int d) {
+    for (int i = 0; i < nsamples; i++) {
+      int64_t extent = input_shape.tensor_shape_span(i)[d];
+      if (static_cast<int64_t>(params_[i].dst_size[d]) != extent ||
+          static_cast<int64_t>(params_[i].src_lo[d]) != 0 ||
+          static_cast<int64_t>(params_[i].src_hi[d]) != extent) {
+        return false;
+      }
+    }
+    return true;
+  };
+
+  int trim_start_ndim = 0;
+  int trim_end_ndim = 0;
+  for (int d = 0; d < spatial_ndim_; d++) {
+    if (!unchanged_dim(d))
+      break;
+    trim_start_ndim++;
+  }
+  for (int d = spatial_ndim_ - 1; d >= 0; d--) {
+    if (!unchanged_dim(d))
+      break;
+    trim_end_ndim++;
+  }
+
+  int new_spatial_ndim = spatial_ndim_ - (trim_start_ndim + trim_end_ndim);
```

Comment:
Is it possible that we have a total no-op specified here and both `trim_start_ndim` and `trim_end_ndim` end up equal to spatial_ndim_?